File: HW1/src/gcd_full.rb
#Maitri Mangal
#CS471 Assignment 1

#! /usr/bin/env ruby

def gcdI(i, j)
	while i != j
		if i > j
			i = i - j
		else
			j = j - i
		end
	end
	return i
end

def gcdF(i, j)
	return gcdF(i-j, j) if i > j
	return i if i == j
	return gcdF(i, j-i)
end 

if ARGV.length != 2 
  puts "gcd_full.rb usage: [NUMBER] [NUMBER]" 
  exit
end

puts "gcdI() : " + gcdI(ARGV[0].to_i, ARGV[1].to_i).to_s
puts "gcdF() : " + gcdF(ARGV[0].to_i, ARGV[1].to_i).to_s

File: HW1/src/gcd_full.py
#Maitri Mangal
#CS471 Assignment 1

#! /usr/bin/env python
import sys

def gcdI(i, j):
	while i != j:
		if i > j:
			i = i - j
		else:
			j = j - i
	return i

#it is called gcdF instead of gcdR in the assignment page
def gcdF(i, j):
	if i > j : return gcdF(i-j, j)
	elif i == j : return i
	else : return gcdF(i, j-i)


if len(sys.argv) != 3:
  print("%s usage: [NUMBER] [NUMBER]" % sys.argv[0])
  exit()

print("gcdI() : "+ str( gcdI(int(sys.argv[1]), int(sys.argv[2]))))
print("gcdF() : "+ str( gcdF(int(sys.argv[1]), int(sys.argv[2]))))
